### PR TITLE
[PF-1384] Add audit log to sshkey controller

### PR DESCRIPTION
**Pull request by yuhuyoyo**

### CODE REVIEW

**Review by ddietterich — APPROVED** [review]
LGTM

File: service/src/main/java/bio/terra/externalcreds/auditLogging/AuditLogEventType.java [hunk 1]
```diff
@@ -6,5 +6,12 @@
   LinkDeleted,
   LinkExpired,
   GetPassport,
-  LinkRefreshed
+  LinkRefreshed,
+  SshKeyPairCreated,
+  SshKeyPairCreationFailed,
+  GetSshKeyPairSucceeded,
+  GetSshKeyPairFailed,
+  SshKayPairDeleted,
```
Comment (dvoet) [inline service/src/main/java/bio/terra/externalcreds/auditLogging/AuditLogEventType.java]:
```suggestion
  SshKeyPairDeleted,
```

File: service/src/test/java/bio/terra/externalcreds/controllers/SshKeyApiControllerTest.java [hunk 2]
```diff
@@ -177,4 +210,13 @@ private void mockSamUser(String accessToken) throws Exception {
...
     when(usersApiMock.getUserStatusInfo()).thenReturn(userStatusInfo);
   }
+
+  private void verifyAuditLogEvent(String sshKeyPairType, AuditLogEventType auditLogEventType) {
+    verify(auditLoggerMock).logEvent(auditLogEventArgumentCaptor.capture());
+    var auditLogEvent = auditLogEventArgumentCaptor.getValue();
+    assertEquals(sshKeyPairType, auditLogEvent.getSshKeyPairType().get().toLowerCase());
+    assertEquals(auditLogEventType, auditLogEvent.getAuditLogEventType());
+
+    Mockito.clearInvocations(auditLoggerMock);
```
Comment (dvoet) [inline service/src/test/java/bio/terra/externalcreds/controllers/SshKeyApiControllerTest.java:220]: I don't think you need to do this. I believe `MockBean`s are reset automatically

**Review by dvoet — APPROVED** [review]
a couple minor changes then 👍

File: service/src/test/java/bio/terra/externalcreds/controllers/SshKeyApiControllerTest.java (hunk 2 above)
Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/externalcreds/controllers/SshKeyApiControllerTest.java:220]: they don't get reset in the same test though. I have one test that do generate, get, delete and verify logging for each command.

**Pull request merged**